src/src_cpp/src_helpers/ortools_utils.cc

```diff
@@ -99,6 +99,121 @@ bool ORTwritemps(operations_research::MPSolver const & solver_p, std::string con
 	return true;
 }
 
+void ORTwritempsPreciseWithCoin(operations_research::MPSolver & solver_p, 
+    std::string const& filename_p) {
+    
+    /*The MPS writer of inner CBC solver will be used
+    At this part, the optimization data in only set in ORTools data, not in CBC Solver
+    To send the data to CBC, the model has to be solved
+    We set a Time limit of 1ms so that the solve, which is only required to send the data
+    is not time consuming */
+    solver_p.set_time_limit(1);
+    solver_p.Solve();
+
+    // We take the underlying solver present in CBC
+    OsiClpSolverInterface* underlying_CBC_solver = 
+        static_cast<OsiClpSolverInterface*>(solver_p.underlying_solver());
+
+    // Getting row names
+    std::vector<std::string> rowNamesVector(solver_p.NumConstraints());
+    int ind = 0;
+    for (auto const& ctr : solver_p.constraints()) {
+        rowNamesVector[ind] = ctr->name();
+        ind++;
+    }
+
+    // Getting column names
+    // A "dummy" variable is added by CBC at index 0 when the problem is solved
+    // It will not be written in MPS but need to be present in names
+    std::vector<std::string> columnNamesVector(solver_p.NumVariables() + 1);
+    columnNamesVector[0] = "dummy";
+    ind = 1;
+    for (auto const& var : solver_p.variables()) {
+        columnNamesVector[ind] = var->name();
+        ind++;
+    }
+
+    int formatType = 1;
+    int numberAcross = 0;
+    double objSense = 1.0;
+    int numberSOS = 0;
+    const CoinSet* setInfo = NULL;
+    ORTwriteMps_CBC_with_names(
+        underlying_CBC_solver,
+        filename_p,
+        formatType,
+        numberAcross,
+        objSense,
+        numberSOS,
+        setInfo,
+        columnNamesVector,
+        rowNamesVector
+    );
+}
+
+int ORTwriteMps_CBC_with_names(OsiClpSolverInterface* solver,
+    std::string const& filename,
+    int formatType,
+    int numberAcross,
+    double objSense,
+    int numberSOS,
+    const CoinSet* setInfo,
+    std::vector<std::string> const& colNamesVec,
+    std::vector<std::string> const& rowNamesVec) 
+{
+    const int numcols = solver->getNumCols();
+    std::shared_ptr<char[]> shared_integrality(new char[numcols]);
+    char* integrality = shared_integrality.get();
+    integrality = CoinCopyOfArray(solver->getColType(false), numcols);
+
+    bool hasInteger = false;
+    for (int i = 0; i < numcols; ++i) {
+        if (solver->isInteger(i)) {
+            hasInteger = true;
+            break;
+        }
+    }
+
+    // Get multiplier for objective function - default 1.0
+    std::shared_ptr<double[]> shared_objective(new double[numcols]);
+    double* objective = shared_objective.get();
+    objective = CoinCopyOfArray(solver->getObjCoefficients(), numcols);
```

Comment:
Same remark,
```cpp
std::shared_ptr<double[]> shared_objective(new double[numcols]);
double* objective = shared_objective.get();
// Change here
CoinCopyN(solver->getObjCoefficients(), numcols, objective);
```
Explanation : `CoinCopyOfArray` allocates a new array, so with the current code, pointer `objective` is overwritten at line 180. In other words, it points to an address unrelated to object `shared_objective(new double[numcols])` and hence will not be freed automatically at the end.

Reply:
Ok, thank you for the verification !